### PR TITLE
#1-4Task.Alexei.Kolgan

### PR DESCRIPTION
**Pull request by kolganem**
Links with Travis-ci about taking tests:

1. - https://travis-ci.com/github/kolganem/js-assignments/builds/173994695
2. - https://travis-ci.com/github/kolganem/js-assignments/builds/174027770
3. - https://travis-ci.com/github/kolganem/js-assignments/builds/174028165 
4. - https://travis-ci.com/github/kolganem/js-assignments/builds/174028579

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -68,8 +68,8 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, John Doe!' => 'John Doe'
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
-function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+function extractNameFromTemplate(value) {    
+    return value.substring(7,value.length-1);
 }
```
Comment (IlyaMokin) [inline task/01-strings-tasks.js:73]: looks good, but please note substring can work with negative numbers also to take symbols from end. In the case `substring(7, -1);` can do the same.

File: task/02-numbers-tasks.js [hunk 2]
```diff
@@ -54,7 +54,8 @@ function getCicleCircumference(radius) {
  *  -3, 3  => 0
  */
 function getAverage(value1, value2) {
-    throw new Error('Not implemented');
+    var res = ((BigInt(value1)+BigInt(value2))/BigInt(2));
+   return res;
```
Comment (IlyaMokin) [inline task/02-numbers-tasks.js:58]: Cool! Unexpected solution =) nodejs  supports BigInt from 10.4 version. And the solution is really dangerous for browsers.
Please try to solve the issue without BigInt also

File: task/03-date-tasks.js [hunk 3]
```diff
@@ -56,7 +56,10 @@ function parseDataFromIso8601(value) {
  *    Date(2015,1,1)    => false
  */
 function isLeapYear(date) {
-   throw new Error('Not implemented');
+   var buff = new Date(date);
+   var year = buff.getFullYear();  
+   return (year % 100 === 0) ? (year % 400 === 0) : (year % 4 === 0)
```
Comment (IlyaMokin) [inline task/03-date-tasks.js:61]: `var` is deprecated. It has no sense to use even for browsers code currently.

**Comment by IlyaMokin** [comment]
I've converted the PR tp draft. @kolganem, please click "ready for review" when it will be ready

**Comment by kolganem** [comment]
Emm... I have several questions. I need to correct your comments. Do I need to do this from my branch? to commit the necessary files and throw the pool request? or something else?

**Comment by kolganem** [comment]
or I don’t need to fix anything in the branch, just read your comments...

File: task/02-numbers-tasks.js (hunk 2 above)
Comment (kolganem) [reply task/02-numbers-tasks.js:58]: fixed

File: task/01-strings-tasks.js (hunk 1 above)
Comment (kolganem) [reply task/01-strings-tasks.js:73]: fixed

Comment (kolganem) [reply task/01-strings-tasks.js:73]: done

File: task/03-date-tasks.js (hunk 3 above)
Comment (kolganem) [reply task/03-date-tasks.js:61]: done

**Comment by IlyaMokin** [comment]
Closing the PR as accepted!
Thanks!

**Pull request closed without merging**